File: spring-security/frontend/ui/src/app/models/owner-property-listing.dto.ts
export interface OwnerPropertyListingDto {
    id: number;
    title: string;
    description?: string;
    price: number;
    propertyType: string;
    fullAddress: string;
    city?: string;
    district?: string;
    latitude?: number;
    longitude?: number;
    formattedAddress?: string;
    rooms?: number;
    bedrooms?: number;
    bathrooms?: number;
    area?: number;
    floor?: number;
    hasBalcony?: boolean;
    securityDeposit?: number;
    availableFrom: string; // Assuming string date format from backend (e.g., "YYYY-MM-DD")
    availableTo?: string; // Assuming string date format
    paymentFrequency?: string;
    minimumStayMonths?: number;
    amenities?: string[];
    contactInfo?: string;
    imageUrls?: string[];
    ownerId: number;
    ownerUsername?: string;
    createdAt: string; // Assuming string date-time format
    updatedAt?: string; // Assuming string date-time format
    listingDate?: string; // Assuming string date-time format
    active: boolean;
} 

File: spring-security/src/main/java/com/example/spring_security/dto/OwnerPropertyListingDTO.java
package com.example.spring_security.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;
import java.util.Set;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class OwnerPropertyListingDTO {
    private Long id;
    private String title;
    private String description;
    private BigDecimal price;
    private String propertyType;
    private String fullAddress;
    private String city;
    private String district;
    private Double latitude;
    private Double longitude;
    private String formattedAddress;
    private Integer rooms;
    private Integer bedrooms;
    private Integer bathrooms;
    private Double area;
    private Integer floor;
    private Boolean hasBalcony;
    private BigDecimal securityDeposit;
    private LocalDate availableFrom;
    private LocalDate availableTo;
    private String paymentFrequency;
    private Integer minimumStayMonths;
    private Set<String> amenities;
    private String contactInfo;
    private List<String> imageUrls;
    private Long ownerId;
    private String ownerUsername; // Or any other relevant user detail
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private LocalDateTime listingDate;
    private boolean active;
} 